core/src/main/java/nl/nn/adapterframework/configuration/IbisContext.java

```diff
@@ -320,14 +321,24 @@ public void load(String configurationName) {
 		}
 	}
 
-	// Create a new configuration through Spring, and explicitly set the ClassLoader before initialising it.
+	/**
+	 * Create a new configuration through Spring, and explicitly set the ClassLoader before initializing it.
+	 * If no ClassLoader or ClassLoader is not IConfigurationClassLoader return an error.
+	 */
 	private Configuration createConfiguration(String name, ClassLoader classLoader) {
+		if(!(classLoader instanceof IConfigurationClassLoader)) {
+			throw new IllegalStateException("no IConfigurationClassLoader set");
+		}
+
 		Configuration bean = (Configuration) getApplicationContext().getAutowireCapableBeanFactory().autowire(Configuration.class, AutowireCapableBeanFactory.AUTOWIRE_BY_NAME, false);
 		bean.setClassLoader(classLoader);
 		return (Configuration) getApplicationContext().getAutowireCapableBeanFactory().initializeBean(bean, name);
 	}
 
-	private void digestClassLoaderConfiguration(ClassLoader classLoader, String currentConfigurationName, ConfigurationException customClassLoaderConfigurationException) {
+	/**
+	 * either ClassLoader is populated or ConfigurationException, but never both!
+	 */
+	private void digestClassLoaderConfiguration(ClassLoader classLoader, String currentConfigurationName, ConfigurationException classLoaderException) {
```

Comment:
Doet dit de classloader configuration digesten? Of een Configuration digesten met een classloader?